### DISCUSSION TITLE
C - How to get textures

### DISCUSSION THREAD

**Discussion by jkvargas** [assimp/assimp#3745] [Q&A] [answered]
Hello,
Maybe someone can give me some tips on how this should be done.
I've seen that there is a list of textures available from the scene.mTextures, one of this properties is pcData which on my understanding holds the binary texture data.
Now, for every material you can call aiGetMaterialTexture which will provide you between other stuff the filename of the texture.
I would like to see a c example on how can you get all the textures for a material with every information about it, including its binary.
I suppose you can link somehow from the aiGetMaterialTexture to the scene.mTextures index...
This will help me on the implementation of russimp, which I desire to refactor the textures side.

Thanks.

**Comment by kimkulling** [comment, marked as answer]
Some of the materials are just external textures which you need to load. Some of the are stored internally as internal textures. The internal textures are called embedded textures. If the pcData property of the texture contains data you can load it directly from there. 
You can find a small example how to load this data here: https://github.com/assimp/assimp/issues/408
